sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs

```diff
@@ -88,13 +88,19 @@ internal static TokenCredential CreateCredential(IConfiguration configuration, T
             var credentialType = configuration["credential"];
             var clientId = configuration["clientId"];
             var tenantId = configuration["tenantId"];
+            var resourceId = configuration["resourceId"];
             var clientSecret = configuration["clientSecret"];
             var certificate = configuration["clientCertificate"];
             var certificateStoreName = configuration["clientCertificateStoreName"];
             var certificateStoreLocation = configuration["clientCertificateStoreLocation"];
 
             if (string.Equals(credentialType, "managedidentity", StringComparison.OrdinalIgnoreCase))
             {
+                if (!string.IsNullOrWhiteSpace(resourceId))
```

Comment:
I put this as the first check before ClientId, because apparently ClientId can be empty as it is an optional parameter to the ManagedIdentityCredential.

Reply:
Should we update the docs to indicate that if both are provided, `resourceId` takes precedence?

Reply:
In Identity, we throw if both are specified.

Reply:
> In Identity, we throw if both are specified.

I considered doing this here, but was hesitant because there is nothing that stopped users from putting resourceId in the config before, so wanted to attempt to handle this gracefully.

Reply:
Chatted offline - will follow same behavior as Identity and update the config key to reduce chance of customers being broken by the change.